rclcpp/include/rclcpp/guard_condition.hpp

```diff
@@ -89,6 +95,15 @@ class GuardCondition
   bool
   exchange_in_use_by_wait_set_state(bool in_use_state);
 
+  /// Adds the guard condition to a waitset
+  /**
+   * This function is thread-safe.
+   * \param[in] wait_set pointer to a wait set where to add the guard condition
+   */
+  RCLCPP_PUBLIC
+  void
+  add_to_wait_set(rcl_wait_set_t * wait_set) const;
```

Comment:
I'm reluctant to add new public API that uses the rcl type here. Ideally, we'd update it to work with the `rclcpp::WaitSet`, as I've said elsewhere.

Since that it out of scope, if we still want this function, I'd like to put it into a free function, in the detail namespace, so it is not a public API, and therefore we can remove it later when we roll out use of the rclcpp WaitSet.

So something like this:

```c++
// rclcpp/include/rclcpp/detail/add_guard_condition_to_rcl_wait_set.hpp
namespace rclcpp
{
namespace detail
{
RCLCPP_PUBLIC
void
add_guard_condition_to_rcl_wait_set(rcl_wait_set_t & wait_set, const rclcpp::GuardCondition & guard_condition);
}  // namespace detail
}  // namespace rclcpp

Reply:
Done in 0d6e8a9